File: sorrisinho_app/lib/classroom/models/discipline.dart
class Discipline {
  final int id;
  final String name;
  final String campus;
  final String period;
  final String code;
  final String classCode;
  final String tpi;
  final String course;

  const Discipline({
    required this.id,
    required this.name,
    required this.campus,
    required this.period,
    required this.code,
    required this.classCode,
    required this.tpi,
    required this.course,
  });

  @override
  bool operator ==(Object other) {
    return other is Discipline &&
        name == other.name &&
        campus == other.campus &&
        period == other.period &&
        classCode == other.classCode &&
        code == other.code &&
        tpi == other.tpi;
  }

  @override
  int get hashCode => id;

  factory Discipline.fromJson(Map<String, dynamic> json) {
    return Discipline(
      id: json['id'],
      name: json['name'],
      campus: json['campus'],
      period: json['period'],
      code: json['code'],
      classCode: json['classCode'],
      tpi: json['tpi'],
      course: json['course'],
    );
  }

  Map<String, dynamic> toJson() {
    return {
      'id': id,
      'name': name,
      'campus': campus,
      'period': period,
      'code': code,
      'classCode': classCode,
      'tpi': tpi,
      'course': course,
    };
  }

  factory Discipline.decodeFromStorage(dynamic storageValue) {
    return Discipline(
      id: storageValue['id'],
      name: storageValue['name'],
      campus: storageValue['campus'],
      period: storageValue['period'],
      code: storageValue['code'],
      classCode: storageValue['classCode'],
      tpi: storageValue['tpi'],
      course: storageValue['course'],
    );
  }
}

File: sorrisinho-api/src/main/java/com/sorrisinho/sorrisinhoapi/business/discipline/Discipline.java
package com.sorrisinho.sorrisinhoapi.business.discipline;

import com.fasterxml.jackson.annotation.JsonIgnoreProperties;
import com.sorrisinho.sorrisinhoapi.business.classroom.Classroom;
import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

import javax.persistence.*;
import java.time.Instant;
import java.util.Date;

@Entity
@Getter
@Setter
@AllArgsConstructor
@NoArgsConstructor
@JsonIgnoreProperties(value={"hibernateLazyInitializer","handler"})
public class Discipline {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(nullable = false)
    private Long id;

    @Column(nullable = false)
    private String name;

    @Column(nullable = false)
    private String campus;

    @Column(nullable = false)
    private String period;

    @Column(nullable = false)
    private String code;

    @Column(nullable = false)
    private String type;

    @Column(nullable = false)
    private String classCode;

    @Column(nullable = false)
    private Integer quarterNumber;

    @Column(nullable = false)
    private Integer quarterYear;

    @Column(nullable = false)
    private Date createdAt = Date.from(Instant.now());

    @Column(nullable = false)
    private String course;

    @Column(nullable = false)
    private String tpi;

    void updateFromAnotherDiscipline(Discipline disciplineToCopy) {
        this.name = disciplineToCopy.getName();
        this.campus = disciplineToCopy.getCampus();
        this.period = disciplineToCopy.getPeriod();
        this.code = disciplineToCopy.getCode();
        this.type = disciplineToCopy.getType();
        this.classCode = disciplineToCopy.getClassCode();
        this.course = disciplineToCopy.getCourse();
        this.tpi = disciplineToCopy.getTpi();

    }


    public Discipline (Integer id) { this.id = Long.valueOf(id); }
}
